lib/screens/login_widget.dart

```diff
@@ -18,22 +17,24 @@ class _LoginWidgetState extends State<LoginWidget> {
   final _passwordController = TextEditingController();
 
   ///
-  void _signUp() async {
+  Future<void> _signUp() async {
     final success = await Services.of(context)
         .authService
         .signUp(_emailController.text, _passwordController.text);
     if (success) {
+      if (!mounted) return;
```

Comment:
Check whether the widget is mounted before the `if success` statement? Will reduce duplication a bit